tests/integration/test_integration.py

```diff
@@ -397,3 +455,41 @@ async def test_given_vault_pki_relation_when_integrate_then_cert_is_provided(
     assert action_output.get("certificate", None) is not None
     assert action_output.get("ca-certificate", None) is not None
     assert action_output.get("csr", None) is not None
+
+@pytest.mark.abort_on_fail
+async def test_given_vault_integrated_with_s3_when_create_backup_then_action_fails(
```

Comment:
Why do we expect failure? and why do we only test for failure?

Reply:
I explained in in the PR notes:

> The integration tests here don't validate end-to-end s3 backup because there is no minio machine charm or any other machine charm that supports the s3 charm relation interface. So we simply validate that the create-backup action fails, as expected with the bad endpoint we provide it. This is likely something we want to improve in the future as the s3 situation on machine charms improve.